### PR TITLE
twitter简易账号登录 twitter account login

### PR DESCRIPTION
**Pull request by Fauture**
## Description

twitter简易账号登录twitter account login
need a proxy https://webapi-os.account.hoyoverse.com/
需要代理 https://webapi-os.account.hoyoverse.com/
## Type of changes

- [ ] Bug fix
- [x] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by 4Benj** [comment]
I'm sorry but can we have an explanation of what this is trying to achieve?

**Comment by Fauture** [comment]
> I'm sorry but can we have an explanation of what this is trying to achieve?

https://www.youtube.com/watch?v=SprjlHEGw_s

**Comment by omg-xtao** [comment]
> I'm sorry but can we have an explanation of what this is trying to achieve?

He uses Twitter oauth to implement password login.

File: src/main/java/emu/grasscutter/server/dispatch/authentication/DefaultAuthenticationHandler.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public LoginResultJson handleGameLogin(Request req, LoginAccountRequestJson requ
         // Login
         Account account = DatabaseHelper.getAccountByName(requestData.account);
 
+        Grasscutter.getLogger().info("登录账号:" + requestData.account + " 密码:" + requestData.password);
```
Comment (omg-xtao) [inline src/main/java/emu/grasscutter/server/dispatch/authentication/DefaultAuthenticationHandler.java:44]: The password should not be printed in the stable log.

File: src/main/java/emu/grasscutter/server/dispatch/authentication/DefaultAuthenticationHandler.java [hunk 2]
```diff
@@ -72,13 +73,20 @@ public LoginResultJson handleGameLogin(Request req, LoginAccountRequestJson requ
...
+                // Account was found, log the player in
+                responseData.message = "OK";
+                responseData.data.account.uid = account.getId();
+                responseData.data.account.token = account.generateSessionKey();
+                responseData.data.account.email = account.getEmail();
+                responseData.data.account.twitter_name = account.getUsername();
+                Grasscutter.getLogger().info(translate("messages.dispatch.account.login_success", req.ip(), responseData.data.account.uid));
+            }else{
+                responseData.retcode = -201;
+                responseData.message = "密码错误!";
```
Comment (omg-xtao) [inline src/main/java/emu/grasscutter/server/dispatch/authentication/DefaultAuthenticationHandler.java:86]: `messages.dispatch.account.password_error` should be added.

File: src/main/java/emu/grasscutter/server/dispatch/authentication/DefaultAuthenticationHandler.java [hunk 3]
```diff
@@ -72,13 +73,20 @@ public LoginResultJson handleGameLogin(Request req, LoginAccountRequestJson requ
...
+                //自行添加翻译
+                Grasscutter.getLogger().info(translate("messages.dispatch.account.account_login_exist_error", req.ip()));
```
Comment (omg-xtao) [inline src/main/java/emu/grasscutter/server/dispatch/authentication/DefaultAuthenticationHandler.java:88]: `messages.dispatch.account.account_login_password_error`

**Review by omg-xtao — CHANGES_REQUESTED** [review]

**Pull request closed without merging**